File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/api/dto/CriarEventoRequest.java
package br.ufpr.sept.so2.modules.presenca.api.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

import java.time.OffsetDateTime;

public record CriarEventoRequest(
        @NotBlank String titulo,
        @NotNull OffsetDateTime inicioEm,
        @NotNull OffsetDateTime fimEm,
        @Positive int cargaHoraria
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/api/dto/CriarEventoRequest.kt
package br.ufpr.sept.so2.modules.presenca.api.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import java.time.OffsetDateTime

data class CriarEventoRequest(
    @field:NotBlank val titulo: String,
    @field:NotNull val inicioEm: OffsetDateTime,
    @field:NotNull val fimEm: OffsetDateTime,
    @field:Positive val cargaHoraria: Int,
)
